application-task-ui/src/main/resources/TaskManager/TaskTemplateList.xml

```diff
@@ -0,0 +1,208 @@
+      <code>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <restricted>0</restricted>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>forbid</cache>
+    </property>
+    <property>
+      <code>define('xwiki-task-template-list-notification-messages', {
+  prefix: 'taskmanager.templateList.notification.',
+  keys: [
+    'inprogress',
+    'done',
+    'fail',
+    'fail.conflict',
+    'fail.unauthorized',
+  ]
+});
+require(['jquery', 'xwiki-l10n!xwiki-task-template-list-notification-messages'], function ($, l10n) {
+  var escapeHTML = function(text) {
+    let textHTMLEscape = new Element('span');
+    textHTMLEscape.textContent = text;
+    return textHTMLEscape.innerHTML;
+  };
+  console.log(l10n)
+  $(document).on('click', '#button-submit-template', function (event) {
+    const endpointURL = new XWiki.Document(
+        XWiki.Model.resolve('TaskManager.CreateTaskTemplateEndpoint', XWiki.EntityType.DOCUMENT)
+    ).getURL('get');
+    event.preventDefault();
+    console.log('ahh');
+    // Check that the template name is not empty.
+    if (event.target.parentElement.reportValidity()) {
+      let notification = new XWiki.widgets.Notification(escapeHTML(l10n.get('inprogress')), 'inprogress');
+      $.ajax(endpointURL + '?' + event.target.parentElement.serialize()).done((data) =&gt; {
```

Comment:
There are nicer ways of making a request with params:

1. `let url = new XWiki.Document(..).getURL('get', $.param({ key: value})`
2. `$.ajax({url: url, data: {key : val}})`